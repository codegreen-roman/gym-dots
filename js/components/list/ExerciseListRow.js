import React from 'react'
import { Exercise } from './Exercise'
import { object } from 'prop-types'
import glamorous from 'glamorous'

const _ExerciseListRow = ({ exercise }) => {
    const Li = glamorous.li({
        padding: 10,
        display: 'flex'
    })
    return (
        <Li>
            <Exercise exercise={exercise} />
        </Li>
    )
}

_ExerciseListRow.propTypes = {
    exercise: object.isRequired
}

export const ExerciseListRow = _ExerciseListRow
