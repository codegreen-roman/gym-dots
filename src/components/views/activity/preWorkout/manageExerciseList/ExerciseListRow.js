import React from 'react'
import { func, string, number } from 'prop-types'
import {
    exerciseRow,
    exerciseDataWrapper,
    exerciseTitle,
    exerciseDetails,
    exerciseElementsWrapper,
    exerciseLine,
    exerciseDot
} from './ExerciseList.glamor'

const onRowClick = () => {} // noop

const _ExerciseListRow = ({
    name,
    weight,
    reps,
    sets,
    exerciseKey,
    onRowClick
}) => {
    return (
        <li {...exerciseRow} data-test={`exercise-row-${exerciseKey}`} onClick={onRowClick}>
            <div {...exerciseElementsWrapper}>
                <div {...exerciseLine} />
                <div {...exerciseDot} />
            </div>
            <div {...exerciseDataWrapper}>
                <div {...exerciseTitle}>
                    {name}
                </div>
                <div {...exerciseDetails} data-test='exercise-details'>
                    {weight} kg x {reps} reps x {sets} sets
                </div>
            </div>
        </li>
    )
}

_ExerciseListRow.defaultProps = {
    onRowClick
}

_ExerciseListRow.propTypes = {
    name: string.isRequired,
    weight: number.isRequired,
    reps: number.isRequired,
    sets: number.isRequired,
    exerciseKey: string.isRequired,
    onRowClick: func.isRequired
}

export const ExerciseListRow = _ExerciseListRow
