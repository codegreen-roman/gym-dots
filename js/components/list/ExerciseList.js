import React from 'react'
import { array, string, func } from 'prop-types'
import glamorous from 'glamorous'

import { ExerciseListRow } from './ExerciseListRow'
import { ExerciseListHeader } from './ExerciseListHeader'

const onOrderChangeClick = () => {} // noop

const _ExerciseList = ({ exercises, title, onOrderChangeClick }) => {

    const Ul = glamorous.ul({
        listStyleType: 'none',
        padding: 0
    })

    return (
        <Ul>
            {exercises.length > 0 ? <ExerciseListHeader title={title} /> : null }
            {exercises.map((exercise, idx) => (
                <ExerciseListRow onRowClick={() => onOrderChangeClick(exercise)} key={idx} exercise={exercise} />
            ))}
        </Ul>
    )
}

_ExerciseList.defaultProps = {
    onOrderChangeClick
}

_ExerciseList.propTypes = {
    exercises: array.isRequired,
    title: string.isRequired,
    onOrderChangeClick: func.isRequired
}

export const ExerciseList = _ExerciseList
