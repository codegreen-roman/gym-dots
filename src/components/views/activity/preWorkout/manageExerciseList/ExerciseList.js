import React from 'react'
import { array, string, func } from 'prop-types'
import {
    exerciseList,
    exerciseListHeader,
    exerciseListHeaderDot,
    exerciseListHeaderText
} from './ExerciseList.glamor'
import { ExerciseListRow } from './ExerciseListRow'
import { compose, not, map, isEmpty } from 'ramda'

export const ExerciseListHeader = ({title}) => {
    return (
        <div data-test='list-title' {...exerciseListHeader}>
            <div {...exerciseListHeaderDot} />
            <div {...exerciseListHeaderText}>{title}</div>
        </div>
    )
}

ExerciseListHeader.propTypes = {
    title: string.isRequired
}

const onOrderChangeClick = () => {} // noop

const _ExerciseList = ({ list, title, onOrderChangeClick }) => {
    const row = (item) => (
        <ExerciseListRow key={item.exerciseKey} {...item} onRowClick={() => onOrderChangeClick(item)} />
    )
    const renderRows = () => map(row, list)
    const notEmpty = compose(not, isEmpty)

    return (notEmpty(list) ?
        (<div>
            <ExerciseListHeader title={title} />
            <ul data-test='list' {...exerciseList}>
                {renderRows()}
            </ul>
        </div>) : null
    )
}

_ExerciseList.defaultProps = {
    onOrderChangeClick
}

_ExerciseList.propTypes = {
    list: array.isRequired,
    title: string.isRequired,
    onOrderChangeClick: func
}

export const ExerciseList = _ExerciseList
