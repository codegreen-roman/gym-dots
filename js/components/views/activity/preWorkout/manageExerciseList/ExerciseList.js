import React from 'react'
import { isEmpty } from 'ramda'
import { array, string, func } from 'prop-types'
import { exerciseList, exerciseListHeader } from './ExerciseList.glamor'
import { ExerciseListRow } from './ExerciseListRow'

const onOrderChangeClick = () => {} // noop

const _ExerciseList = ({ list, title, onOrderChangeClick }) => {
    if (isEmpty(list)) {
        return null
    }
    return (
        <div>
            <span data-test='list-title' {...exerciseListHeader}>
                {title}
            </span>
            <ul data-test='list' {...exerciseList}>
                {list.map((itm, idx) =>
                    <ExerciseListRow
                        key={idx}
                        {...itm}
                        onRowClick={() => onOrderChangeClick(itm)}
                    />
                )}
            </ul>
        </div>
    )
}

_ExerciseList.defaultProps = {
    onOrderChangeClick
}

_ExerciseList.propTypes = {
    list: array.isRequired,
    title: string.isRequired,
    onOrderChangeClick: func
}

export const ExerciseList = _ExerciseList
