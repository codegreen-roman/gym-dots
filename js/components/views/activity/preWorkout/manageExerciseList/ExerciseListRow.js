import React from 'react'
import { func, string, number } from 'prop-types'
import { exerciseTitle, exerciseDetails } from './ExerciseList.glamor'

const onRowClick = () => {} // noop

const _ExerciseListRow = ({ name, weight, reps, sets, onRowClick, exerciseId }) => {
    return (
        <li data-test={`exercise-row-${exerciseId}`} onClick={onRowClick}>
            <h3 {...exerciseTitle}>
                {name}
            </h3>
            <span {...exerciseDetails}>
                {weight} kg x {reps} reps x {sets} sets
            </span>
        </li>
    )
}

_ExerciseListRow.defaultProps = {
    onRowClick
}

_ExerciseListRow.propTypes = {
    name: string.isRequired,
    weight: number.isRequired,
    reps: number.isRequired,
    sets: number.isRequired,
    onRowClick: func.isRequired
}

export const ExerciseListRow = _ExerciseListRow
