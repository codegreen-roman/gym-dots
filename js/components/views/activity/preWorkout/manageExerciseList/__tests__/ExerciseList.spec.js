import React from 'react'
import { shallow, mount } from 'enzyme'
import { ExerciseList } from '../ExerciseList'
import { ExerciseListRow } from '../ExerciseListRow'
import { ExerciseListHeader } from '../ExerciseList'

describe('ExerciseList component', () => {
    const props = {
        title: 'Some title',
        list: [
            {
                exerciseKey: 'xxy',
                restTime: 90,
                name: 'Push-ups',
                sets: 5,
                reps: 20,
                weight: 0,
                results: []
            },
            {
                exerciseKey: 'xyy',
                restTime: 30,
                name: 'Australian pull-ups',
                sets: 5,
                reps: 12,
                weight: 0,
                results: []
            }
        ]
    }

    const setup = props => {
        const actions = {
            onOrderChangeClick: jest.fn()
        }

        const component = shallow(<ExerciseList {...props} {...actions} />)
        const wrapper = mount(<ExerciseList {...props} {...actions} />)

        return {
            component,
            wrapper,
            actions,
            props,
            title: component.find(ExerciseListHeader),
            list: component.find('[data-test="list"]'),
            row: component.find(ExerciseListRow)
        }
    }

    it('should render without crashing', () => {
        const { component } = setup(props)
        expect(component).toMatchSnapshot()
    })

    it('should have list of type ul', () => {
        const { list } = setup(props)
        expect(list.node.type).toBe('ul')
    })

    it('should have length of rows to be 2', () => {
        const { row } = setup(props)
        expect(row.length).toBe(2)
    })

    it('should have list header', () => {
        const { title } = setup(props)
        expect(title).toMatchSnapshot()
    })

    it('should have list header with title matching text of prop title', () => {
        const { title } = setup(props)
        expect(title.props().title).toMatch(/^Some title/)
    })

    describe('When list is empty and has length 0', () => {
        let props
        beforeEach(() => {
            props = {
                onOrderChangeClick: jest.fn(),
                title: '',
                list: []
            }
        })

        it('should not render the component and return null', () => {
            const { wrapper } = setup(props)
            expect(wrapper.find('[data-test="list"]')).toMatchSnapshot()
        })

        it('should have length of rows to be 0', () => {
            const { row } = setup(props)
            expect(row.length).toBe(0)
        })
    })

    describe('Clicking on a row', () => {

        it('should match the snapshot', () => {
            const { wrapper } = setup(props)
            expect(wrapper).toMatchSnapshot()
        })

        it('should call onOrderChangeClick for every row - click', () => {
            const { wrapper, actions: { onOrderChangeClick } } = setup(props)
            const rows = wrapper.find('li')

            for (let r = 0; r < rows.length; r++) {
                rows.at(r).simulate('click')
            }

            expect(onOrderChangeClick).toHaveBeenCalledTimes(rows.length)
        })

        it('should call onOrderChangeClick with list item object', () => {
            const { wrapper, actions: { onOrderChangeClick } } = setup(props)
            const row = wrapper.find('li').at(0)

            row.simulate('click')

            expect(onOrderChangeClick).toHaveBeenCalledWith({
                exerciseKey: 'xxy',
                restTime: 90,
                name: 'Push-ups',
                sets: 5,
                reps: 20,
                weight: 0,
                results: []
            })

        })

    })
})
