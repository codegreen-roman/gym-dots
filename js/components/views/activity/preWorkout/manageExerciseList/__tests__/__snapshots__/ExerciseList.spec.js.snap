// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExerciseList component Clicking on a row should match the snapshot 1`] = `
<_ExerciseList
  list={
    Array [
      Object {
        "exerciseKey": "xxy",
        "name": "Push-ups",
        "reps": 20,
        "restTime": 90,
        "results": Array [],
        "sets": 5,
        "weight": 0,
      },
      Object {
        "exerciseKey": "xyy",
        "name": "Australian pull-ups",
        "reps": 12,
        "restTime": 30,
        "results": Array [],
        "sets": 5,
        "weight": 0,
      },
    ]
  }
  onOrderChangeClick={[Function]}
  title="Some title"
>
  <div>
    <span
      data-css-3atj9v=""
      data-test="list-title"
    >
      Some title
    </span>
    <ul
      data-css-1d8iqpl=""
      data-test="list"
    >
      <_ExerciseListRow
        exerciseKey="xxy"
        name="Push-ups"
        onRowClick={[Function]}
        reps={20}
        restTime={90}
        results={Array []}
        sets={5}
        weight={0}
      >
        <li
          data-test="exercise-row-xxy"
          onClick={[Function]}
        >
          <h3
            data-css-17h8928=""
          >
            Push-ups
          </h3>
          <span
            data-css-1jge3ly=""
          >
            0
             kg x 
            20
             reps x 
            5
             sets
          </span>
        </li>
      </_ExerciseListRow>
      <_ExerciseListRow
        exerciseKey="xyy"
        name="Australian pull-ups"
        onRowClick={[Function]}
        reps={12}
        restTime={30}
        results={Array []}
        sets={5}
        weight={0}
      >
        <li
          data-test="exercise-row-xyy"
          onClick={[Function]}
        >
          <h3
            data-css-17h8928=""
          >
            Australian pull-ups
          </h3>
          <span
            data-css-1jge3ly=""
          >
            0
             kg x 
            12
             reps x 
            5
             sets
          </span>
        </li>
      </_ExerciseListRow>
    </ul>
  </div>
</_ExerciseList>
`;

exports[`ExerciseList component When list is empty and has length 0 should not render the component and return null 1`] = `undefined`;

exports[`ExerciseList component should render without crashing 1`] = `
<div>
  <span
    data-css-3atj9v=""
    data-test="list-title"
  >
    Some title
  </span>
  <ul
    data-css-1d8iqpl=""
    data-test="list"
  >
    <_ExerciseListRow
      exerciseKey="xxy"
      name="Push-ups"
      onRowClick={[Function]}
      reps={20}
      restTime={90}
      results={Array []}
      sets={5}
      weight={0}
    />
    <_ExerciseListRow
      exerciseKey="xyy"
      name="Australian pull-ups"
      onRowClick={[Function]}
      reps={12}
      restTime={30}
      results={Array []}
      sets={5}
      weight={0}
    />
  </ul>
</div>
`;
